### PR TITLE
Tailscale VPN

### PR DESCRIPTION
**Pull request by mstormi**
Personal auto-VPN based on WireGuard ... see docs files in PR

Fixes: #1045 
Fixes: #1114

Signed-off-by: Markus Storm <markus.storm@gmx.net>

### CODE REVIEW

**Comment by mstormi** [comment]
Install the 'tailscale' branch via clonebranch= if you want to test the feature before a merge.

**Comment by holgerfriedrich** [comment]
Hi Markus, I am sorry, I won't try this VPN solution. Just a few general remarks:

- I prefer free software. Wireguard is open source, Tailscaile is not. It looks modern and well maintained, the background reads good. Mesh sounds good, but not really a big advantage for a private OH install? Well, I cannot judge....
- Second thing about the admin key.... I did not get why it is downloaded? We don't deploy it into /boot dir of a new image? (btw: naming could be changed to give a hint that this is a public key....
- You may have forgotten to remove `sources.list.d/tailscale.list` on uninstall (I may have overlooked it)

File: functions/menu.bash
Comment (ecdye) [inline functions/menu.bash:135]: Where is the option to remove tailscale from the menu? Does the wireguard remove remove tailscale as well?

File: functions/vpn.bash
Comment (ecdye) [inline functions/vpn.bash:191]: Remove the admin ssh key as well

File: openhabian-setup.sh
Comment (ecdye) [inline openhabian-setup.sh:113]: Add a check in `install_tailscale` that will cancel the installation at that point if the user has not input the needed values in `openhabian.conf`  otherwise we are bloating the install process with unrequested packages by installing tailscale then canceling afterwards.

File: functions/vpn.bash
Comment (ecdye) [inline functions/vpn.bash:225]: Add a check to ensure that exim is installed if interactive.

File: functions/vpn.bash
Comment (ecdye) [inline functions/vpn.bash:180]: Unused

File: functions/auth.bash
Comment (ecdye) [inline functions/auth.bash:110]: Add a check to ensure that tailscale is installed before running.

File: functions/auth.bash
Comment (ecdye) [inline functions/auth.bash:47]: Why change to 8?

**Review by ecdye — CHANGES_REQUESTED** [review]

**Comment by mstormi** [comment]
> Wireguard is open source, Tailscaile is not.

You're mistaken there, it is in fact: https://github.com/tailscale/tailscale
Just downloading the compiled package from their repo like we do for e.g. Zulu Java as well.
I have even been thinking to build a private instance but that turned out to become a pretty time consuming adventure so I abandoned the idea.
Given the service is free for personal use it's IMHO reasonable to use that.

> but not really a big advantage for a private OH install? Well, I cannot judge....

It's not about mesh but the whole point why I want that in is about firewalling.
I have actually been trying to piece together what's needed for a remote install and got stuck (you need DynDNS and some UPnP magic to open firewall ports I could not even find any program for, let alone that we would need to write lengthy how-tos). It was foreseeable to become a mess of variants with lots of pitfalls users might hit along the way, depending on the type of router they use.
When I then stumbled across that one during my search, I tried it and it worked and convinced me right away.
Please give it a second chance and at least a try to be able to give a judgement.

> Second thing about the admin key.... I did not get why it is downloaded

It's not a master key for us (maintainers/image builders) but an option to auto-download an installer's key.
Say I wanted to setup an openHABian instance in my mom's house, then she's the user but I'm the installer (or "admin"), and I would want my key to be present right from the beginning so my mom just needs to place the hardware and turn it on then I can login via VPN and proceed with configuring the system.
I've been thinking of several alternatives such as to generate a key pair and send them via mail but ended up with this one as the IMHO most simple and safe way.
If you know a more elegant way let me know.

> You may have forgotten to remove sources.list.d/tailscale.list on uninstall

yep, will add

File: functions/auth.bash
Comment (mstormi) [reply functions/auth.bash:47]: I'm no fan of lengthy passwords. Being forced to invent one makes you use passes you can remember (hence way more insecure).
Had a discourse on that with @ThomDietrich a while back and he didn't mind changing it down to 8. I thought I had already done that back then now saw I didn't. If you believe it's inherently more secure to stick with 10 we can do.

File: functions/auth.bash
Comment (mstormi) [reply functions/auth.bash:110]: I'd think that is not necessarily related ? You could want not to use tailscale but say port forwarding and use the admin key nonetheless.

File: functions/vpn.bash
Comment (mstormi) [reply functions/vpn.bash:191]: hmm, really? see https://github.com/openhab/openhabian/pull/1174#discussion_r507951840

File: functions/menu.bash
Comment (mstormi) [reply functions/menu.bash:135]: no, I added it

File: openhabian-setup.sh
Comment (mstormi) [reply openhabian-setup.sh:113]: ok like so?

File: functions/vpn.bash
Comment (mstormi) [reply functions/vpn.bash:239]: done

File: functions/vpn.bash
Comment (mstormi) [reply functions/vpn.bash]: what is unused ?

File: functions/auth.bash
Comment (ecdye) [reply functions/auth.bash:47]: I have no real issue with it, I was just wondering.

File: functions/auth.bash
Comment (ecdye) [reply functions/auth.bash:110]: But this function directly calls tailscale cli if I recall correctly, right?

File: functions/vpn.bash
Comment (ecdye) [reply functions/vpn.bash]: queryText, right?

File: openhabian-setup.sh
Comment (ecdye) [reply openhabian-setup.sh:113]: Sure

File: functions/vpn.bash
Comment (mstormi) [reply functions/vpn.bash]: No I think it was in use. Anyway, I just changed that.

File: functions/auth.bash
Comment (mstormi) [reply functions/auth.bash:110]: duh, right. I moved adding the key to console to setup_tailscale()

File: NEWS.md
Comment (ecdye) [inline NEWS.md:15]: Capitalize offline.

File: docs/NEWSLOG.md
Comment (ecdye) [inline docs/NEWSLOG.md:10]: Capitalize offline.

**Review by ecdye — APPROVED** [review]
LGTM, DNT.

**Comment by mstormi** [comment]
Thanks Ethan.

I'd like to encourage both of you to at least try this stuff so at least someone other than me can confirm the code works - even if you dislike this or do not want to use it.

**Pull request merged**
